core/controllers/voice_artist.py

```diff
@@ -126,3 +145,92 @@ def post(self, unused_exploration_id):
+    def get(self, purpose):
+        """Handles GET requests."""
+        if purpose == feconf.VOICEOVER_APPLICATION_REVIEW:
+            voiceover_applications = (
+                voiceover_services.get_reviewable_voiceover_applications(
+                    self.user_id))
+        elif purpose == feconf.VOICEOVER_APPLICATION_STATUS:
+            status = self.request.get('status')
+            voiceover_applications = (
+                voiceover_services.get_user_submitted_voiceover_applications(
+                    self.user_id, status=status))
+        else:
+            raise self.PageNotFoundException
+
+        self.values = {
+            'voiceover_applications': [
+                v_a.to_dict() for v_a in voiceover_applications]
+        }
+        self.render_json(self.values)
+
+
+class VoicoverApplicationHandler(base.BaseHandler):
+    """Handler for the voiceover application."""
+
+    @acl_decorators.can_review_voiceover_application
+    def put(self, voiceover_application_id):
+
+        action = self.payload.get('action')
+
+        if action == suggestion_models.ACTION_TYPE_ACCEPT:
+            voiceover_services.accept_voiceover_application(
+                voiceover_application_id, self.user_id)
+        elif action == suggestion_models.ACTION_TYPE_REJECT:
+            review_message = self.payload.get('review_message')
+            voiceover_services.reject_voiceover_application(
+                voiceover_application_id, self.user_id, review_message)
+        else:
+            raise self.InvalidInputException('Invalid action.')
+
+        self.render_json({})
+
+    @acl_decorators.can_submit_voiceover_application
+    def post(self):
+        raw_audio_file = self.request.get('raw_audio_file')
+
+        target_type = self.payload.get('target_type')
+        target_id = self.payload.get('target_id')
+        language_code = self.payload.get('language_code')
+        voiceover_content = self.payload.get('voiceover_content')
+        filename = (
+            self.user_id + utils.generate_random_string(6) + '.mp3')
+
+        try:
+            _save_audio_file(
+                raw_audio_file, filename,
+                feconf.ENTITY_TYPE_VOICEOVER_APPLICATION, target_id,
+                self.user_id)
+            voiceover_services.create_new_voiceover_application(
+                target_type, target_id, language_code, voiceover_content,
+                filename, self.user_id)
+        except Exception as e:
+            raise self.InvalidInputException(e)
+
+        self.render_json({})
+
+
+class VoiceoverApplicationTextHandler(base.BaseHandler):
+    """Handler for voiceover application content."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_submit_voiceover_application
+    def get(self, target_type, target_id):
+        language_code = self.request.get('language_code')
+
+        try:
+            text = voiceover_services.get_text_to_create_voiceover_application(
```

Comment:
Can we rename this to get_text_to_voiceover? I think that that's what this function is doing.

Reply:
`get_text_to_create_voiceover_application` returns a voiceover which is required by voiceover application i.e, it will return a specific text which is used in the voiceover application.

Should we change this to get_text_for_voiceover_application? (I'm fine with `get_text_to_voiceover` as well!)

Reply:
get_text_to_voiceover sounds simpler and I think it conveys the entire meaning. Let's go with that.

Reply:
> `get_text_to_create_voiceover_application` returns a voiceover which is required by voiceover application i.e, it will return a specific text which is used in the voiceover application.

It returns a voiceover? I thought it returned text which needs to be voiceovered. (Sorry I missed this comment in prev review)